test/proxy_test.lua

```diff
@@ -0,0 +1,82 @@
+local t = require('luatest')
+local replica_set = require('luatest.replica_set')
+local server = require('luatest.server')
+local proxy = require('luatest.replica_proxy')
+
+local fiber = require('fiber')
+
+local g = t.group('proxy', {
+    {is_paused = false},
+    {is_paused = true},
+})
+
+local function tarantool_version_ge(major, minor, patch)
+    local version = require('tarantool').version
+    version = version:split('.')
+    local major1 = tonumber(version[1]:match('%d+'))
+    local minor1 = tonumber(version[2]:match('%d+'))
+    local patch1 = tonumber(version[3]:match('%d+'))
+    if major1 ~= major then
+        return major1 > major
+    elseif minor1 ~= minor then
+        return minor1 > minor
+    else
+        return patch1 >= patch
+    end
+end
+
+g.before_all(function(cg)
+    -- Proxy only works on tarantool 2.10+
+    t.run_only_if(tarantool_version_ge(2, 10, 0),
+                  'Proxy works on Tarantool 2.10+')
+    cg.replica_set = replica_set:new{}
+    cg.box_cfg = {
+        replication_timeout = 0.1,
+        replication = {
+            server.build_listen_uri('server2_proxy'),
+        },
+    }
+    cg.server1 = cg.replica_set:build_and_add_server{
+        alias = 'server1',
+        box_cfg = cg.box_cfg,
+    }
+    cg.box_cfg.replication = nil
+    cg.server2 = cg.replica_set:build_and_add_server{
+        alias = 'server2',
+        box_cfg = cg.box_cfg,
+    }
+    cg.proxy = proxy:new{
+        client_socket_path = server.build_listen_uri('server2_proxy'),
+        server_socket_path = server.build_listen_uri('server2'),
```

Comment:
When you rebase your patch on master, this test will fail. Now you should pass `cg.replica_set.id` to make it work.

Reply:
Thanks! Rebased on master and fixed the test:
```diff
diff --git a/test/proxy_test.lua b/test/proxy_test.lua
index 73e5fdf..eacff2e 100644
--- a/test/proxy_test.lua
+++ b/test/proxy_test.lua
@@ -29,28 +29,28 @@ g.before_all(function(cg)
     -- Proxy only works on tarantool 2.10+
     t.run_only_if(tarantool_version_ge(2, 10, 0),
                   'Proxy works on Tarantool 2.10+')
-    cg.replica_set = replica_set:new{}
+    cg.rs = replica_set:new{}
     cg.box_cfg = {
         replication_timeout = 0.1,
         replication = {
-            server.build_listen_uri('server2_proxy'),
+            server.build_listen_uri('server2_proxy', cg.rs.id),
         },
     }
-    cg.server1 = cg.replica_set:build_and_add_server{
+    cg.server1 = cg.rs:build_and_add_server{
         alias = 'server1',
         box_cfg = cg.box_cfg,
     }
     cg.box_cfg.replication = nil
-    cg.server2 = cg.replica_set:build_and_add_server{
+    cg.server2 = cg.rs:build_and_add_server{
         alias = 'server2',
         box_cfg = cg.box_cfg,
     }
     cg.proxy = proxy:new{
-        client_socket_path = server.build_listen_uri('server2_proxy'),
-        server_socket_path = server.build_listen_uri('server2'),
+        client_socket_path = server.build_listen_uri('server2_proxy', cg.rs.id),
+        server_socket_path = server.build_listen_uri('server2', cg.rs.id),
     }
     t.assert(cg.proxy:start{force = true}, 'Proxy is started')
-    cg.replica_set:start{}
+    cg.rs:start{}
 end)
 
 g.test_server_disconnect_is_noticed = function(cg)
@@ -78,5 +78,5 @@ g.test_server_disconnect_is_noticed = function(cg)
 end
 
 g.after_all(function(cg)
-    cg.replica_set:drop()
+    cg.rs:drop()
 end)
```